assignments/callbacks.js

```diff
@@ -1,55 +1,75 @@
 // Create a higher order function and invoke the callback function to test your work. You have been provided an example of a problem and a solution to see how this works with our items array.  Study both the problem and the solution to figure out the rest of the problems.
 
-const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
+const items = [ 'Pencil', 'Notebook', 'yo-yo', 'Gum' ];
 
-/* 
+//Given this problem:
 
-  //Given this problem: 
-  
-  function firstItem(arr, cb) {
-    // firstItem passes the first item of the given array to the callback function.
-  }
+// function firstItem(arr, cb) {
+// 	// firstItem passes the first item of the given array to the callback function.
+// }
 
-  // Potential Solution:
+// // Potential Solution:
 
-  // Higher order function using "cb" as the call back
-  function firstItem(arr, cb) {
-    return cb(arr[0]);
-  }
-
-  // Function invocation 
-  firstItem(items, function(first) {
-    console.log(first)
-  });
-
-*/
+// // Higher order function using "cb" as the call back
+// function firstItem(arr, cb) {
+// 	return cb(arr[0]);
+// }
 
+// // Function invocation
+// firstItem(items, function(first) {
+// 	console.log(first);
+// });
 
 function getLength(arr, cb) {
-  // getLength passes the length of the array into the callback.
+	return cb([ arr.length ]);
+	// getLength passes the length of the array into the callback.
 }
 
+getLength(items, function(length) {
+	console.log(length);
+});
+
 function last(arr, cb) {
-  // last passes the last item of the array into the callback.
+	// last passes the last item of the array into the callback.
+	return cb(arr[arr.length - 1]);
 }
+last(items, function(lastitem) {
+	console.log(lastitem);
+});
 
 function sumNums(x, y, cb) {
-  // sumNums adds two numbers (x, y) and passes the result to the callback.
+	// sumNums adds two numbers (x, y) and passes the result to the callback.
+	return cb(x, y);
+}
```

Comment:
When you are working with Callbacks, you want to pass the function to the higher order function.  For example:

```sh
function sumNums(x,y,cb) {
     return cb(x + y);
}

sumNums(1, 2, sum => {
    console.log(sum) // Result : 3
});
```

Look at this example to answer the multiply one.